src/main/scala/is/hail/expr/FunctionRegistry.scala

```diff
@@ -2067,43 +2067,125 @@ object FunctionRegistry {
 
   registerLambdaAggregator("takeBy", (f: (Any) => Any, n: Int) => new TakeByAggregator[Int](f, n),
```

Comment:
I put them into a variable because there are special examples for integral vs. floating-point numbers and no examples for Strings.